oak_functions/examples/weather_lookup/module/src/lib.rs

```diff
@@ -24,41 +24,100 @@ use serde::Deserialize;
+            .map_err(|err| format!("could not deserialize request as JSON: {:?}", err))?;
+        log!("parsed request: {:?}\n", request).unwrap();
+
+        // Fetch the index entry, containing all the concatenated keys.
+        let index = oak_functions::storage_get_item("index".as_bytes())
+            .map_err(|err| format!("could not get item: {:?}", err))?
+            .ok_or("could not find index item")?;
+
+        // Build a `Location` object from the client request.
+        let request_location = Location {
+            latitude_millidegrees: (request.latitude_degrees * 1000.0) as i32,
+            longitude_millidegrees: (request.longitude_degrees * 1000.0) as i32,
+        };
+
+        // Find the closest key by linearly scanning the keys from the index and computing their
+        // distance to the client request location.
+        let best_key = index
+            .chunks(8)
+            .min_by_key(|key| {
+                let location = Location::from_bytes(key);
+                location.distance(&request_location)
+            })
+            .ok_or("could not find nearest location")?;
+        log!("nearest location key: {:?}\n", best_key).unwrap();
+
+        let best_value = oak_functions::storage_get_item(best_key)
+            .map_err(|err| format!("could not get item: {:?}", err))?
+            .ok_or("could not find item with key")?;
+        log!("nearest location value: {:?}\n", best_value).unwrap();
+
+        best_value
     };
 
     let response = result.unwrap_or_else(|err| err.as_bytes().to_vec());
 
     // Write the response.
     oak_functions::write_response(&response).expect("Couldn't write the response body.");
 }
+
+/// We use a fixed point representation (`i32`, instead of `f32`) for the location coordinates in
+/// order to be more efficient at parsing it from bytes, and also to compute distance via arithmetic
+/// operations.
+#[derive(Debug, Eq, PartialEq)]
+struct Location {
+    latitude_millidegrees: i32,
+    longitude_millidegrees: i32,
+}
+
+impl Location {
+    fn from_bytes(c: &[u8]) -> Location {
+        let mut latitude_millidegrees_bytes = [0; 4];
+        latitude_millidegrees_bytes.copy_from_slice(&c[0..4]);
+        let mut longitude_millidegrees_bytes = [0; 4];
+        longitude_millidegrees_bytes.copy_from_slice(&c[4..8]);
+
+        Location {
+            latitude_millidegrees: i32::from_be_bytes(latitude_millidegrees_bytes),
+            longitude_millidegrees: i32::from_be_bytes(longitude_millidegrees_bytes),
+        }
+    }
+
+    #[cfg(test)]
+    fn to_bytes(&self) -> Vec<u8> {
+        [
+            self.latitude_millidegrees.to_be_bytes(),
+            self.longitude_millidegrees.to_be_bytes(),
+        ]
+        .concat()
+    }
+
+    fn distance(&self, other: &Location) -> i32 {
+        // Convert to `i64` in order to avoid overflow when squaring.
+        (((self.latitude_millidegrees as i64 - other.latitude_millidegrees as i64).pow(2)
```

Comment:
Perhaps add a TODO to resolve the shortcomings, such as wrapping longitudes?

Reply:
Done.